src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1436,3 +1464,178 @@ def write_sorting(
+        E.g., calling
+            write_recording(recording=my_recording_extractor, nwbfile=my_nwbfile)
+        will result in the appropriate changes to the my_nwbfile object.
+        If neither 'nwbfile_path' nor 'nwbfile' are specified, an NWBFile object will be automatically generated
+        and returned by the function.
+    metadata: dict, optional
+        Metadata dictionary with information used to create the NWBFile when one does not exist or overwrite=True.
+        The "Ecephys" section of metadata is also used to create electrodes and electrical series fields.
+    recording: BaseRecording, optional
+        If the waveform_extractor is 'recordingless', this argument needs to be passed to save electrode info.
+        Otherwise, electrodes info is not added to the nwb file.
+    overwrite: bool, optional
+        Whether or not to overwrite the NWBFile if one exists at the nwbfile_path.
+        The default is False (append mode).
+    verbose: bool, optional
+        If 'nwbfile_path' is specified, informs user after a successful write operation.
+        The default is True.
+    unit_ids: list, optional
+        Controls the unit_ids that will be written to the nwb file. If None (default), all
+        units are written.
+    write_electrical_series: bool, optional
+        If True, the recording object associated to the WaveformExtractor is written as an electrical series.
+    write_electrical_series_kwargs: dict, optional
+        Keyword arguments to control the `write_recording()` function in case write_electrical_series=True
+    property_descriptions: dict
+        For each key in this dictionary which matches the name of a unit
+        property in sorting, adds the value as a description to that
+        custom unit column.
+    skip_properties: list of str
+        Each string in this list that matches a unit property will not be written to the NWBFile.
+    skip_features: list of str
+        Each string in this list that matches a spike feature will not be written to the NWBFile.
+    write_as: str (optional, defaults to 'units')
+        How to save the units table in the nwb file. Options:
+        - 'units' will save it to the official NWBFile.Units position; recommended only for the final form of the data.
+        - 'processing' will save it to the processing module to serve as a historical provenance for the official table.
+    units_name : str (optional, defaults to 'units')
+        The name of the units table. If write_as=='units', then units_name must also be 'units'.
+    units_description : str (optional)
+    """
+    if metadata is None:
+        metadata = dict()
+    assert (
+        nwbfile_path is None or nwbfile is None
+    ), "Either pass a nwbfile_path location, or nwbfile object, but not both!"
+    if nwbfile is not None:
+        assert isinstance(nwbfile, pynwb.NWBFile), "'nwbfile' should be a pynwb.NWBFile object!"
+
```

Comment:
I think these assertions (1539 to 1543) are not necessary here, the `make_or_load_nwbfile` context manager already has those checks.

Reply:
ok